enhancements/storage/csi-ebs-one-zone.md

```diff
@@ -0,0 +1,271 @@
+  - @gnufied
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - None
+creation-date: 2025-07-09
+last-updated: 2025-07-09
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/STOR-2365
+see-also:
+  - "/enhancements/csi-efs-operator.md"
+replaces:
+superseded-by:
+---
+
+# AWS EFS CSI driver support for One Zone volumes
+
+## Summary
+
+Bring support of One Zone volumes into the AWS EFS CSI driver.
+
+## Motivation
+
+One Zone EFS volumes bring lower price and higher speed, at the price of less availability. Only machines in the same availability zone (AZ) as the volume can mount One Zone volume efficiently. Cross-AZ mounting is still possible, however, AWS will charge cross-AZ traffic + the speed will suffer.
+
+We want cluster admins to be able to use One Zone volumes in OpenShift.
+
+### User Stories
+
+* As a cluster admin, I can manually create a PersistentVolume (PV) and PersistentVolumeClaim (PVC) for an One Zone EFS volume, so it can be used only by Pods on nodes in the same AZ as the volume.
+  * Here we assume the cluster admin sets the correct `spec.nodeAffinity` in the PV, so the scheduler knows where the volume is.
+* As a cluster admin, I can create a StorageClass that dynamically provisions PVs from an already existing One Zone EFS volume. Such volumes will be usable only in the same AZ as the volume.
+  * As a cluster admin, I must also ensure that the whole cluster is in a single AWS AZ. The dynamically provisioned PVs do not have the correct `spec.nodeAffinity`, see [an upstream issue](https://github.com/kubernetes-sigs/aws-efs-csi-driver/issues/1153).
+* As a cluster admin, I can upgrade the driver without any explicit re-configuration and my regional PVs will keep working.
+
+### Goals
+
+* Be on par with the upstream CSI driver and support One Zone EFS volumes.
+* Keep supporting regional EFS volumes out of the box, including upgrade from older CSI driver versions - all regional PVs keep working after the upgrade without any action needed.
+
+### Non-Goals
+
+* Solve One Zone EFS volumes topology.
+  * **Dynamic provisioning of One Zone volumes in a multi-AZ cluster is unsupported.**  The CSI driver does not support topology and thus the scheduler does not know where the volumes are and will schedule pods to wrong zones. See [an upstream issue](https://github.com/kubernetes-sigs/aws-efs-csi-driver/issues/1153).
+  * Manual provisioning is supported, as long as the cluster admin provides `spec.nodeAffinity` in PVs, so the scheduler knows where the volumes are.
```

Comment:
Why manual provisioning in non-goals section if this is supported?

Reply:
Removed + rephrased a bit